server/client.go

```diff
@@ -4992,6 +4992,11 @@ func (c *client) doTLSHandshake(typ string, solicit bool, url *url.URL, tlsConfi
 				}
 			}
 		}
+	} else if !c.matchesPinnedCert(pCerts) {
+		err = ErrConnectionClosed
+	}
+
+	if err != nil {
 		if kind == CLIENT {
```

Comment:
I would have kept this in the Handshake() error statement, unless you want to have matchedPinnedCert() failure to also be reported as a TLS handshake error, but then, maybe you want to introduce a new error that indicates that it is due to pinned_cert not matching..

Reply:
have a separate error now